paimon-core/src/main/java/org/apache/paimon/metrics/AbstractMetricGroup.java

```diff
@@ -0,0 +1,187 @@
+    }
+
+    /**
+     * Creates and registers a new {@link org.apache.paimon.metrics.Counter}.
+     *
+     * @param name name of the counter
+     * @return the created counter
+     */
+    public Counter counter(String name) {
+        return counter(name, new SimpleCounter());
+    }
+
+    /**
+     * Registers a {@link org.apache.paimon.metrics.Counter}.
+     *
+     * @param name name of the counter
+     * @param counter counter to register
+     * @param <C> counter type
+     * @return the given counter
+     */
+    public <C extends Counter> C counter(String name, C counter) {
+        addMetric(name, counter);
+        return counter;
+    }
+
+    /**
+     * Registers a new {@link org.apache.paimon.metrics.Gauge}.
+     *
+     * @param name name of the gauge
+     * @param gauge gauge to register
+     * @param <T> return type of the gauge
+     * @return the given gauge
+     */
+    public <T, G extends Gauge<T>> G gauge(String name, G gauge) {
+        addMetric(name, gauge);
+        return gauge;
+    }
+
+    /**
+     * Registers a new {@link Histogram} with Paimon.
+     *
+     * @param name name of the histogram
+     * @param histogram histogram to register
+     * @param <H> histogram type
+     * @return the registered histogram
+     */
+    public <H extends Histogram> H histogram(String name, H histogram) {
+        addMetric(name, histogram);
+        return histogram;
+    }
+
+    /**
+     * Adds the given metric to the group and registers it at the registry, if the group is not yet
+     * closed, and if no metric with the same name has been registered before.
+     *
+     * @param metricName the name to register the metric under
+     * @param metric the metric to register
+     */
+    protected void addMetric(String metricName, Metric metric) {
+        if (metric == null) {
+            LOG.warn(
+                    "Ignoring attempted registration of a metric due to being null for name {}.",
+                    metricName);
+            return;
+        }
+        // add the metric only if the group is still open
+        synchronized (this) {
+            if (!isClosed()) {
+                switch (metric.getMetricType()) {
+                    case COUNTER:
+                    case GAUGE:
+                    case HISTOGRAM:
+                        metrics.put(metricName, metric);
+                        break;
+                    default:
+                        LOG.warn(
+                                "Cannot add unknown metric type {}. This indicates that the paimon "
+                                        + "does not support this metric type.",
+                                metric.getClass().getName());
+                }
+            }
+        }
+    }
+
+    @Override
+    public Map<String, Metric> getMetrics() {
+        Map<String, Metric> metricsMap = new HashMap<>();
+        synchronized (this) {
+            metrics.forEach((k, v) -> metricsMap.put(k, v));
+        }
```

Comment:
Use `ConcurrentHashMap`. No need for `synchronized`.

Reply:
I think `ConcurrentHashMap` won't guarantee the sequence of iterator and `put` operation right? Or there is no need to guarantee the sequence of iterator and the `put` operation you think?

Reply:
It doesn't matter if the metrics reporter reports a metric group slightly earlier or later, so there is no need to maintain strict consistency here.